dashboard/src/t5gweb/libtelco5g.py

```diff
@@ -240,191 +240,281 @@ def create_cards(cfg, new_cases, action="none"):
-            notifieduser = assignee.get("notifieduser", "true")
-            # Add the user as a watcher only if they want to be notified
-            if notifieduser == "true" and case:
-                logging.warning(
-                    f"Adding watcher {assignee['jira_user']} to case {case}"
-                )
-                add_watcher_to_case(cfg, case, assignee["jira_user"], token)
-            else:
-                logging.warning(
-                    f"Not adding watcher {assignee['jira_user']} to case {case}"
-                )
-
-        priority = portal2jira_sevs[cases[case]["severity"]]
-        full_description = (
-            "This card was automatically created from the Case Dashboard Sync Job.\r\n"
-            + "\r\n"
-            + "This card was created because it had a severity of "
-            + cases[case]["severity"]
-            + "\r\nThe account for the case is "
-            + cases[case]["account"]
-            + "\r\nThe case had an internal status of: "
-            + cases[case]["status"]
-            + "\r\n\r\n*Description:* \r\n\r\n"
-            + cases[case]["description"]
-            + "\r\n"
+        context["jira_conn"].add_issues_to_sprint(
+            context["sprint"].id, [previous_issue.key]
         )
-        summary = case + ": " + cases[case]["problem"]
-        card_info = {
-            "project": {"key": cfg["project"]},
-            "issuetype": {"name": cfg["type"]},
-            "components": [{"name": cfg["component"]}],
-            "priority": {"name": priority},
-            "labels": cfg["labels"],
-            "summary": summary[:253] + ".." if len(summary) > 253 else summary,
-            "description": (
-                full_description[:253] + ".."
-                if len(full_description) > 253
-                else full_description
-            ),
-        }
+        context["jira_conn"].transition_issue(previous_issue, "11")
+        context["jira_conn"].add_comment(
+            previous_issue,
+            f"Case {case} seems to have been reopened. The dashboard found "
+            "this card linked to the case and reopened it automatically.",
+        )
+        return True
+    return False
+
+
+def _determine_assignee(case, cases, cfg):
+    """Determine who should be assigned to the case"""
+    """Generated by: Cursor"""
+    if not cfg["team"]:
+        return None
+
+    # Try to match by account
+    for member in cfg["team"]:
+        for account in member["accounts"]:
+            if account.lower() in cases[case]["account"].lower():
+                member["displayName"] = member["name"]
+                return member
+
+    # No match found, assign randomly
+    last_choice = redis_get("last_choice")
+    assignee = get_random_member(cfg["team"], last_choice)
+    redis_set("last_choice", json.dumps(assignee))
+    assignee["displayName"] = assignee["name"]
+    return assignee
+
+
+def _add_case_watcher(cfg, case, jira_user, token):
```

Comment:
Rather than adding this function, could we add the logging statement to the existing function? `add_watcher_to_case()`?

Reply:
good catch! done